www/index.php

```diff
@@ -75,139 +64,172 @@
+            "(`token`, `provider_token_id`, `provider_configuration_id`, `user_id`) VALUES " .
+            "(:token, :providerTokenId, :providerConfigurationId, :userId)"
+        );
+        $securityAuthenticationTokenStatement->bindValue(':token', session_id(), \PDO::PARAM_STR);
+        $securityAuthenticationTokenStatement->bindValue(':providerTokenId', $providerTokenId, \PDO::PARAM_INT);
+        $securityAuthenticationTokenStatement->bindValue(
+            ':providerConfigurationId',
+            $configurationId,
+            \PDO::PARAM_INT
+        );
+        $securityAuthenticationTokenStatement->bindValue(':userId', $centreon->user->user_id, \PDO::PARAM_INT);
+        $securityAuthenticationTokenStatement->execute();
+
+        manageRedirection($centreon, $pearDB);
+        return;
+    }
+}
+
+/**
+ * Update centreon base uri in index.html
+ */
+function updateCentreonBaseUri(): void
+{
+    $requestUri = htmlspecialchars($_SERVER['REQUEST_URI']) ?: '/centreon/';
+    $basePath = '/' . trim(explode('index.php', $requestUri)[0], "/") . '/';
+    $basePath = str_replace('//', '/', $basePath);
+    $indexHtmlPath = './index.html';
+    $indexHtmlContent = file_get_contents($indexHtmlPath);
+
+    // update base path only if it has changed
+    if (!preg_match('/.*<base\shref="' . preg_quote($basePath, '/') . '">/', $indexHtmlContent)) {
+        $indexHtmlContent = preg_replace(
+            '/(^.*<base\shref=")\S+(">.*$)/s',
+            '${1}' . $basePath . '${2}',
+            $indexHtmlContent
+        );
+
+        file_put_contents($indexHtmlPath, $indexHtmlContent);
+    }
+}
+
+/**
+ * Redirect to user default page
+ *
+ * @param Centreon $centreon
+ * @param CentreonDB $pearDB
+ */
+function manageRedirection(Centreon $centreon, CentreonDB $pearDB): void
+{
+    $headerRedirection = "./main.php";
+    $argP = filter_var(
+        $_POST['p'] ?? $_GET["p"] ?? null,
+        FILTER_VALIDATE_INT
+    );
+    if ($argP !== false) {
+        $headerRedirection .= "?p=" . $argP;
+        foreach ($_GET as $parameter => $value) {
+            if (! in_array($parameter, AUTOLOGIN_FIELDS)) {
+                $sanitizeParameter = htmlspecialchars($parameter);
+                $sanitizeValue = filter_input(INPUT_GET, $parameter);
+                if (! empty($sanitizeParameter) && $sanitizeValue !== false) {
+                    $headerRedirection .= '&' . $parameter . '=' . $value;
                 }
             }
-            header("Location: " . $headerRedirection);
+        }
+    } elseif (isset($centreon->user->default_page) && $centreon->user->default_page != '') {
+        // get more details about the default page
+        $stmt = $pearDB->prepare(
+            "SELECT topology_url, is_react FROM topology WHERE topology_page = ? LIMIT 0, 1"
+        );
+        $pearDB->execute($stmt, [$centreon->user->default_page]);
+
+        if ($stmt->rowCount() && ($topologyData = $stmt->fetch()) && $topologyData['is_react']) {
```

Comment:
Although it is functional, any value (!= 0) in is_react will validate the test.
It would be better to verify either that the value is different from 0 or equal to 1